src/autoscaler/metricsforwarder/forwarder/syslog_emitter.go

```diff
@@ -0,0 +1,103 @@
+package forwarder
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/metricsforwarder/config"
+	"code.cloudfoundry.org/app-autoscaler/src/autoscaler/models"
+	"code.cloudfoundry.org/go-loggregator/v9/rpc/loggregator_v2"
+	"code.cloudfoundry.org/lager/v3"
+	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress"
+	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress/syslog"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type SyslogEmitter struct {
+	logger lager.Logger
+	Writer egress.WriteCloser
+}
+
+func NewSyslogEmitter(logger lager.Logger, conf *config.Config) (MetricForwarder, error) {
+	var writer egress.WriteCloser
+	var protocol string
+
+	tlsConfig, _ := conf.SyslogConfig.TLS.CreateClientConfig()
+
+	netConf := syslog.NetworkTimeoutConfig{
+		WriteTimeout: time.Second,
+		DialTimeout:  100 * time.Millisecond,
+	}
+
+	if conf.SyslogConfig.TLS.CACertFile != "" {
+		protocol = "syslog-tls"
+	} else {
+		protocol = "syslog"
+	}
+
+	syslogUrl, _ := url.Parse(fmt.Sprintf("%s://%s:%d", protocol, conf.SyslogConfig.ServerAddress, conf.SyslogConfig.Port))
+
+	logger.Info("using-syslog-url", lager.Data{"url": syslogUrl})
+
+	hostname, _ := os.Hostname()
+
+	binding := &syslog.URLBinding{
+		URL:      syslogUrl,
+		Hostname: hostname,
+	}
+
+	syslogEgressCounter := prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "autoscaler",
+		Subsystem: "metricsforwarder",
+		Name:      "syslog_metrics_egress_counter",
+		Help:      "count of metrics sent to syslog",
+	})
```

Comment:
We need to _somehow_ expose this counter to the outside world in order to register it in in the same place where all the registrations happen currently:

https://github.com/cloudfoundry/app-autoscaler-release/blob/a009b0236cd919a741e92d69d9dbf37126f3f7f0/src/autoscaler/metricsforwarder/cmd/metricsforwarder/main.go#L74

https://github.com/cloudfoundry/app-autoscaler-release/blob/a009b0236cd919a741e92d69d9dbf37126f3f7f0/src/autoscaler/metricsforwarder/cmd/metricsforwarder/main.go#L94-L101